// @welovedevs/ui/styles/utils/styles_utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getComponentColor = exports.getHexFromTheme = exports.flexUtils = void 0;
var theme_1 = require("../theme");
// Flex utils
exports.flexUtils = {
    center: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
};
var getHexFromTheme = function (inputTheme, color, shade) {
    var _a, _b;
    if (color === void 0) { color = 'primary'; }
    if (shade === void 0) { shade = 500; }
    var theme = Object.keys(inputTheme || {}).length ? inputTheme : theme_1.DEFAULT_THEME;
    // @ts-ignore
    return (_b = (_a = theme.palette) === null || _a === void 0 ? void 0 : _a[color]) === null || _b === void 0 ? void 0 : _b[shade];
};
exports.getHexFromTheme = getHexFromTheme;
var getComponentColor = function (active, color, disabled, defaultValue) {
    if (defaultValue === void 0) { defaultValue = '#fff'; }
    if (disabled) {
        return '#c0c0c0';
    }
    if (active && color) {
        return color;
    }
    return defaultValue;
};
exports.getComponentColor = getComponentColor;
