// @welovedevs/ui/checkbox/checkbox_styles.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.styles = void 0;
var styles_utils_1 = require("../styles/utils/styles_utils");
var center = styles_utils_1.flexUtils.center;
var styles_1 = require("@material-ui/core/styles");
exports.styles = styles_1.createStyles({
    container: __assign({ height: 24, width: 24, position: 'relative', borderRadius: 5, margin: 10, padding: 3, cursor: 'pointer', overflow: 'hidden' }, center),
    raised: {
        backgroundColor: 'currentColor',
        '& $checkIcon': {
            fill: '#fff',
        },
    },
    outlined: {
        border: "1px solid currentColor",
    },
    disabled: {
        cursor: 'not-allowed',
    },
    isRadio: {
        '&, & > $brightLayer': {
            borderRadius: '50%',
        },
    },
    brightLayer: {
        height: '100%',
        width: '100%',
        position: 'absolute',
        top: 0,
        left: 0,
        backgroundColor: '#fff',
        zIndex: 1,
    },
    checkIcon: {
        height: '100%',
        width: 'auto',
        fill: 'currentColor',
    },
    input: {
        height: '100%',
        width: '100%',
        position: 'absolute',
        top: 0,
        left: 0,
        opacity: 0,
        cursor: 'inherit',
        padding: 0,
        margin: 0,
        zIndex: 2,
    },
});
