// @welovedevs/ui/banner/banner_styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.styles = void 0;
var banner_data_1 = require("./banner_data");
var styles_1 = require("@material-ui/core/styles");
var styles = function (theme) {
    return styles_1.createStyles({
        container: function (_a) {
            var _b, _c, _d, _e;
            var type = _a.type;
            var color = (_c = (_b = banner_data_1.BANNER_DATA === null || banner_data_1.BANNER_DATA === void 0 ? void 0 : banner_data_1.BANNER_DATA[type || 'warning']) === null || _b === void 0 ? void 0 : _b.color) !== null && _c !== void 0 ? _c : 'primary';
            return {
                // @ts-ignore
                color: ((_e = (_d = theme.palette) === null || _d === void 0 ? void 0 : _d[color]) !== null && _e !== void 0 ? _e : theme.palette.primary)[500],
                width: '100%',
                padding: '25px 40px',
                position: 'relative',
                display: 'flex',
                alignItems: 'center',
                '&::before': {
                    height: '100%',
                    width: '100%',
                    position: 'absolute',
                    top: 0,
                    left: 0,
                    backgroundColor: 'currentColor',
                    opacity: 0.15,
                    borderRadius: 5,
                    zIndex: 0,
                    content: "''",
                },
                '& > *': {
                    zIndex: 1,
                },
            };
        },
        iconContainer: {
            display: 'flex',
            marginRight: 30,
            '& > svg': {
                height: 60,
                width: 60,
                '& > g > path:last-child': {
                    fill: 'currentColor',
                },
            },
        },
        '@media screen and (max-width: 550px)': {
            iconContainer: {
                display: 'none',
            },
        },
        '@media screen and (max-width: 400px)': {
            container: {
                padding: 25,
            },
        },
    });
};
exports.styles = styles;
