// @welovedevs/ui/old_carousel/carousel_styles.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.styles = void 0;
var createStyles_1 = __importDefault(require("@material-ui/styles/createStyles"));
var palettes_1 = require("../../app/palettes");
var js_1 = require("../../../style/js");
exports.styles = createStyles_1.default({
    heading: __assign({ height: 340, maxHeight: '40vh', position: 'relative' }, js_1.flexUtils.center),
    video: {
        zIndex: 2,
        width: '100%',
        height: '100%',
    },
    frontImage: {
        width: '100%',
        height: '100%',
        objectFit: 'contain',
        zIndex: 2,
        padding: 10,
    },
    backgroundBlur: {
        position: 'absolute',
        height: '100%',
        width: '100%',
        top: 0,
        left: 0,
        objectFit: 'cover',
    },
    body: {
        maxHeight: '40vh',
        minHeight: 180,
        flex: 1,
        overflowY: 'auto',
        backgroundColor: 'white',
        margin: [20, 20, 30, 20],
    },
    bodyWrapper: __assign(__assign({}, js_1.flexUtils.center), { justifyContent: 'space-around', flexDirection: 'column' }),
    buttons: {
        display: 'flex',
        alignItems: 'center',
    },
    dialogPaper: {
        width: '100%',
        maxWidth: 600,
        overflow: 'visible',
        '& .slick-dots': {
            bottom: -50,
        },
    },
    swipeIcon: {
        position: 'absolute',
        bottom: 5,
        left: 5,
        animation: 'swipe-icon-movement 1s ease-in-out infinite alternate',
        transformOrigin: 'bottom center',
    },
    iNeedHelp: {
        cursor: 'pointer',
        position: 'absolute',
        bottom: 5,
        right: 5,
        fontStyle: 'italic',
        fontSize: 12,
    },
    carousel: {
        flex: 1,
    },
    title: {
        padding: [0, 20],
        marginBottom: 20,
    },
    reverseButton: {
        '& > svg': {
            transform: 'scale(-1)',
        },
    },
    nextButton: {
        right: -80,
    },
    previousButton: {
        left: -80,
    },
    navigateButton: __assign(__assign({ top: 'calc(50% - 25px)', position: 'absolute', backgroundColor: palettes_1.secondary[500], height: 50, width: 50, boxShadow: '0 20px 67px 0 rgba(0,0,0,.3)', cursor: 'pointer', borderRadius: '50%' }, js_1.flexUtils.center), { '& > svg': {
            height: 20,
            width: 'auto',
        } }),
    carouselStep: {
        position: 'relative',
        minWidth: 300,
        display: 'flex',
        flexDirection: 'column',
        borderRadius: 4,
        overflow: 'hidden',
    },
    carouselStepFullScreen: {
        borderRadius: 0,
        height: '100vh',
    },
});
